bundle/src/main/java/com/adobe/acs/commons/models/injectors/impl/SharedValueMapValueInjectorImpl.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2018 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.models.injectors.impl;
+
+import com.adobe.acs.commons.models.injectors.annotation.SharedValueMapValue;
+import com.adobe.acs.commons.wcm.PageRootProvider;
+import com.adobe.acs.commons.wcm.properties.shared.SharedComponentProperties;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.wcm.api.Page;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Property;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+import org.apache.sling.models.spi.DisposalCallbackRegistry;
+import org.apache.sling.models.spi.Injector;
+import org.osgi.framework.Constants;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+@Service
+@Property(name = Constants.SERVICE_RANKING, intValue = 4500)
+public class SharedValueMapValueInjectorImpl implements Injector {
+    @Reference
+    private PageRootProvider pageRootProvider;
+
+    @Override
+    public String getName() {
+        return "shared-component-properties-valuemap";
+    }
+
+    @Override
+    public Object getValue(Object adaptable, String name, Type declaredType, AnnotatedElement element, DisposalCallbackRegistry callbackRegistry) {
+        // sanity check
+        if (element.getAnnotation(SharedValueMapValue.class) == null) {
```

Comment:
For some reason this function gets called by with `name = modelFactory` and `name = slingModelFilter`.  I dont know enough about how these injectors work to understand why those values are triggering this code in the first place, but this is my way of early exit when it does happen.

Reply:
If the intention is to *require* the annotation, this is OK. The idea with injectors is that unless the model developer explicitly defines the source (either through the `@Source` annotation or an injector-specific annotation), all available injectors are checked. In this case, I can see two possible approaches.

1. What you have now -- require the annotation.
2. Don't require the annotation and make sure that this injector is called *after* the regular ValueMap injector (which has a ranking of 2000, per https://sling.apache.org/documentation/bundles/models.html, so you're already good there since this injector's ranking is 4500).

I'd suggest to go with #1 (as you have) as it would be the least surprising *and* the construction of the merged properties has some cost associated with it (not a lot of cost, but some cost).

Reply:
Interesting that it checks all annotations.  I never knew or expected that.  Might be best to stick with as-is to avoid unexpected side effects.